docs/zh_CN/2019-03-19-core-js-3-babel-and-a-look-into-the-future.md

```diff
@@ -0,0 +1,490 @@
+- `Map.prototype.toJSON` 和 `Set.prototype.toJSON` 也是很久前被拒绝了
+- 不必要并且被错误添加的迭代器方法：`CSSRuleList`，`MediaList`，`StyleSheetList`。
+
+#### 不再有非标准、非提案的功能
+
+许多年前，我开始写一个库，他是我的JavaScript程序的核心：这个库包函 polyfills 和一些我需要的工具函数。一段时间后，这个库以 `core-js` 命名发布。我认为现在大多数 `core-js` 用户不需要非标准的 `core-js` 功能，他们大多已经在早期版本移除了，现在是时候将剩余部分从 `core-js` 中移除。从这个版本开始，`core-js` 可以被称为  polyfill 了。
+
+### 包、入口和模块名字
+
+一个issue里提了 `core-js` 包的很大（ ~2MB ），有很多重复文件。因为这个原因，`core-js` 分成了3个包：
+- [`core-js`](https://www.npmjs.com/package/core-js) 定义全局的 polyfills。（ ~500KB，[压缩并且 gzipped 处理后 40KB](https://bundlephobia.com/result?p=core-js@3.0.0-beta.20) ）
+- [`core-js-pure`](https://www.npmjs.com/package/core-js-pure)，提供了不污染全局变量的 polyfills。它和 `core-js@2` 中的 `core-js/library` 相当。（~440KB）
+- [`core-js-bundle`](https://www.npmjs.com/package/core-js-bundle)：定义了全局填充的打包版本
+
+`core-js` 的早期版本中，稳定的 ECMAScript 功能和 ECMAScript 提案的 polyfill 模块化需要分别加 `es6.` 和 `es7.` 前缀。这是在 2014 年做的决定，那时将 ES6 之后的所有功能都视为 ES7。在 `core-js@3` 中所有稳定的 ECMAScript 功能都增加 `es.` 前缀，ECMAScript 提案增加 `esnext.` 前缀。
+
+几乎所有的 CommonJS 入口都改变了。`core-js@3` 相比于 `core-js@2` 有更多的入口：这带来的最大限度的灵活性，使你能够仅仅引入你的应用需要的依赖。
+
+这里是一些例子关于如何使用新的入口：
+
+```js
+// 使用 `core-js` 全部功能打补丁：
+import "core-js";
+// 仅仅使用稳定的 `core-js` 功能 - ES 和 web 标准：
+import "core-js/stable";
+// 仅仅使用稳定的 ES 功能
+import "core-js/es";
+
+// 如果你想用 `Set` 的补丁
+// 所有 `Set`- ES 提案中，相关的功能：
+import "core-js/features/set";
+// 稳定的 `Set` ES 功能和来自web标准的功能
+// （DOM 集合迭代器）
+import "core-js/stable/set";
+// 只有 `Set` 所需的稳定的 ES 功能
+import "core-js/es/set";
+// 与上面一致，但不会污染全局命名空间
+import Set from "core-js-pure/features/set";
+import Set from "core-js-pure/stable/set";
+import Set from "core-js-pure/es/set";
+
+
+// 仅仅为需要的方法打补丁
+import "core-js/feature/set/intersection";
+import "core-js/stable/queque-microtask";
+import "core-js/es/array/from";
+
+// 为 reflect metadata 提案打补丁
+import "core-js/proposals/reflect-metadata";
+// 为所有 stage 2+ 的提案打补丁
+import "core-js/stage/2";
+```
+
+### 其他重要的变化
+
+`core-js` polyfill 能够 [配置侵入等级](https://github.com/zloirock/core-js/blob/master/README.md#configurable-level-of-aggressiveness)。如果你认为有些情境 `core-js` 功能检测侵入性太强，原生实现对你来说已经足够，或者一个错误的实现没有被 `core-js` 检测到，你可以修改 `core-js` 的默认行为。
+
+如果无法安装规范的每个细节实现某个功能，`core-js` 增加了一个 `.sham` 属性，例如，IE11中 `Symbol.sham` 是 `true`。
+
+不再有 LiveScript! 当我开始写 `core-js` 时，我主要使用的是 [LiveScript](http://livescript.net/) ；一段时间后，我用 JavaScript 重写了全部的 polyfills 。在 `core-js@2` 中测试和帮助的工具函数仍然使用 LiveScript ：它是非常有趣的像 CoffeeScript 一样的语言，有强大的语法糖使你能够写非常紧凑的代码，但是它几乎已经死了。除此之外，它也是为 `core-js` 贡献的屏障，因为大多数 `core-js` 用户不知道这个语言。`core-js@3` 测试和工具函数使用现代 ES 语法：它将成为为 `core-js` 贡献的好时机🙂。
+
+对于大多数用户，为了优化 `core-js` 导入，我建议使用 [babel](#Babel)。当然，有些情况下 [`core-js-builder`](http://npmjs.com/package/core-js-builder) 仍然有用。现在它支持 `target` 参数，使用带有目标引擎的[`浏览器列表`](https://github.com/browserslist/browserslist) 查询 - 你能够创建一个 bundle，仅仅包含目标引擎需要的 polyfills。对于这种情况，我做了 [`core-js-compat`](https://www.npmjs.com/package/core-js-compat)，更多关于它的信息，你能够从 [这篇文章的 `@babel/preset-env` 部分](#babelpreset-env)了解到。
```

Comment:
```suggestion
对于大多数用户，为了优化 `core-js` 导入，我建议使用 [babel](#Babel)。当然，有些情况下 [`core-js-builder`](http://npmjs.com/package/core-js-builder) 仍然有用。现在它支持 `target` 参数，使用带有目标引擎的 [`browserslist`](https://github.com/browserslist/browserslist) 查询 - 你能够创建一个 bundle，仅仅包含目标引擎需要的 polyfills。对于这种情况，我做了 [`core-js-compat`](https://www.npmjs.com/package/core-js-compat)，更多关于它的信息，你能够从 [这篇文章的 `@babel/preset-env` 部分](#babelpreset-env)了解到。
```

`browserslist` should be considered as a well-known proper noun and shouldn't be translated.